linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/Analytics.java

```diff
@@ -0,0 +1,224 @@
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Report events to
+ * <a href='https://g3doc.corp.google.com/company/teams/concord/integration/quick_start.md?cl=head'>Clearcut<a/>.
+ */
+final class Analytics {
+
+  // TODO is this a good value?
+  private static final int DELAY = 773; // a prime number
+
+  private static final Logger logger = Logger.getLogger(Analytics.class.getName());
+
+  private static final String FIRELOG_COLLECTION_URL =
+      "https://firebaselogging-pa.googleapis.com/v1/firelog/legacy/log";
+
+  // Don't change the value; this name is used as an originating "application" of usage metrics.
+  private static final String METRICS_NAME = "cloud-opensource-java-linkage-monitor";
+  
+  // for Concord Clearcut
+  private static final String CONSOLE_TYPE = "LINKAGE_MONITOR";
+  
+  private static Analytics instance;
+
+  private final String collectionUrl;
+
+  private final ConcurrentLinkedQueue<PingEvent> pingEventQueue;
+  
+  private int sequencePosition = 0;
+
+  private final boolean sendAnalytics;
+  
+  @VisibleForTesting
+  Analytics(String collectionUrl, 
+      ConcurrentLinkedQueue<PingEvent> concurrentLinkedQueue, boolean sendAnalytics) {
+    this.collectionUrl = collectionUrl;
+    this.pingEventQueue = concurrentLinkedQueue;
+    this.sendAnalytics = sendAnalytics;
+  }
+
+  // TODO any synchronization issues here? triple check that this makes sense.
+  private void start() {
+    Runnable emptyQueue = new Runnable() {
+      @Override
+      public void run() {
+        // drain the queue; GA can handle our traffic without buffering
+        PingEvent event = pingEventQueue.poll();
+        while (event != null) {
+          if (sendAnalytics && event != null) {
+            ping(event);
+          }
+          event = pingEventQueue.poll();
+        }
+      }
+    };
+    
+    ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(1);
+    
+    // getExitingScheduledExecutorService doesn't block app from exiting
+    ScheduledExecutorService scheduler = MoreExecutors.getExitingScheduledExecutorService(
+        executor, 10, TimeUnit.SECONDS);
+
+    scheduler.scheduleAtFixedRate(emptyQueue, DELAY, DELAY, TimeUnit.MILLISECONDS);
+  }
+
+  static synchronized Analytics getInstance(boolean sendAnalytics) {
+    if (instance == null) {
+      String collectionUrl = null;
+      if (!PlaceholderConstants.FIRELOG_API_KEY.startsWith("@")) {
```

Comment:
I don't see `FIRELOG_API_KEY` any more. I remember something was to replace compile-time value. Where did it go?

Reply:
Already checked in https://github.com/GoogleCloudPlatform/cloud-opensource-java/blob/master/linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/PlaceholderConstants.java

Reply:
Thank you.